public/plugin.ts

```diff
@@ -264,6 +320,23 @@ export class ObservabilityPlugin
     coreRefs.toasts = core.notifications.toasts;
     coreRefs.chrome = core.chrome;
     coreRefs.application = core.application;
+    coreRefs.dataSources = startDeps.data.dataSources;
+
+    const { dataSourceService, dataSourceFactory } = startDeps.data.dataSources;
+
+    // register all s3 datasources
+    dataSourceFactory.registerDataSourceType(S3_DATASOURCE_TYPE, S3DataSource);
+    core.http.get(`${DATACONNECTIONS_BASE}`).then((s3DataSources) => {
+      s3DataSources.map((s3ds) => {
+        dataSourceService.registerDataSource(
+          dataSourceFactory.getDataSourceInstance(S3_DATASOURCE_TYPE, {
+            name: s3ds.name,
+            type: s3ds.connector.toLowerCase(),
+            metadata: s3ds,
+          })
+        );
+      });
+    });
```

Comment:
Where is this being registered/why are you doing this?

Reply:
Is this to get it to show up in management datasources plugin...?

Reply:
Do you need to add some dependency into `opensearch_dashboards.yml` to make sure the datasources is available at this point?